PostBoard.Api/Controllers/BirthdayController.cs

```diff
@@ -1,37 +1,36 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
 using PostBoard.Api.Models;
 using PostBoard.Api.Validators;
-using System.Collections.Generic;
-using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.EntityFrameworkCore;
+using PostBoard.Api.Services.Contracts;
 
 namespace PostBoard.Api.Controllers;
 
 [Route("api/birthdays")]
 [ApiController]
 public class BirthdayController : ControllerBase
 {
-    private readonly PostBoardContext _dbcontext;
-
-    public BirthdayController(PostBoardContext dbContext)
+    private readonly IBirthdayService _birthdayService;
+ 
+    public BirthdayController(IBirthdayService birthdayService)
     {
-        _dbcontext = dbContext;
+        _birthdayService = birthdayService;
     }
 
     [HttpGet]
     public async Task<IActionResult> GetAllAsync()
     {
-        var birthdays = await _dbcontext.Birthdays.ToListAsync();
+        var birthdays = await _birthdayService.GetAllAsync();
         
         return Ok(birthdays);
     }
 
     [HttpGet]
-    [Route("{id:int}")]
+    [Route("{id:int}", Name = "GetByIdAsync")]
     public async Task<IActionResult> GetByIdAsync([FromRoute] int id)
     {
-        var birthday = await _dbcontext.Birthdays.FirstOrDefaultAsync(b => b.Id == id);
+        var birthday = await _birthdayService.GetByIdAsync(id);
 
         if (birthday == null)
         {
```

Comment:
While using method `_birthdayService.GetByIdAsync(id)`, you don't have to check if birthday is null in controller.
You have to check if the entity is null in Service.

Reply:
And if it's null, you have to throw exception.